pkg/test/e2e/kafka/kafka_test.go

```diff
@@ -0,0 +1,152 @@
+	if err != nil {
+		return fmt.Errorf("postFLPDeploy:wait.For error:%v", err)
+	}
+	fmt.Printf("====> Done.\n")
+	return nil
+}
+
+func postStrimziDeploy(ctx context.Context, cfg *envconf.Config, namespace string) error {
+	fmt.Printf("====> Waiting for Strimzi and Kafka Deployment\n")
+	client, err := cfg.NewClient()
+	if err != nil {
+		return fmt.Errorf("postStrimziDeploy:NewClient error:%v", err)
+	}
+	dep := appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{Name: "strimzi-cluster-operator", Namespace: namespace},
+	}
+	// wait for the deployment to finish becoming available
+	err = wait.For(conditions.New(client.Resources()).DeploymentConditionMatch(&dep, appsv1.DeploymentAvailable, v1.ConditionTrue), wait.WithTimeout(time.Minute*10))
+	if err != nil {
+		return fmt.Errorf("postStrimziDeploy:wait.For error:%v", err)
+	}
+
+	// wait for kafka to be ready
+	res := gexe.RunProc(fmt.Sprintf("kubectl wait kafka/my-cluster --for=condition=Ready --timeout=20m -n %s", namespace))
+	if !res.IsSuccess() {
+		return fmt.Errorf("postStrimziKafkaDeploy error: %v", res.Err())
+	}
+	fmt.Printf("====> Done.\n")
+	return nil
+}
+
+func TestKafka_Basic(t *testing.T) {
+	pipelineFeature := features.New("FLP/kafka").WithLabel("env", "dev").
+		Setup(func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			return ctx
+		}).
+		Assess("Kafka working as expected", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			client, err := cfg.NewClient()
+			if err != nil {
+				t.Fatal(err)
+			}
+			dep := appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{Name: "flowlogs-pipeline", Namespace: cfg.Namespace()},
+			}
+			// wait for the deployment to finish becoming available
+			err = wait.For(conditions.New(client.Resources()).DeploymentConditionMatch(&dep, appsv1.DeploymentAvailable, v1.ConditionTrue), wait.WithTimeout(time.Minute*3))
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			// coreV1Client from current context in kubeconfig
+			coreV1Client, err := e2e.GetCoreV1Client(cfg.KubeconfigFile())
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			var pods corev1.PodList
+			err = client.Resources(cfg.Namespace()).List(context.TODO(), &pods)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			logs := e2e.LogsFromPods(pods, coreV1Client, cfg.Namespace())
+			fmt.Print(logs)
+			startExist := strings.Contains(logs, "Starting flowlogs-pipeline")
+			if !startExist {
+				t.Fatal("can't find flowlogs-pipeline start message")
+			}
+			errorsExist := strings.Contains(logs, "ERROR")
+			if errorsExist {
+				t.Fatal("flowlogs-pipeline reports errors ")
+			}
+
+			command := "kubectl get kafka my-cluster -o=jsonpath='{.status.listeners[?(@.type==\"external\")].bootstrapServers}'"
+			kafkaAddr := test.RunCommand(command)
+			// strip the quotation marks
+			kafkaServer = kafkaAddr[1 : len(kafkaAddr)-1]
```

Comment:
`kafkaServer` is a variable defined in `kafka_utils.go`. Do we want to add characters to its name to indicate that it's a package global variable rather than a local variable?

Reply:
Change to `theKafkaServer`